packages/workflows/src/definition/cart/create-cart.ts

```diff
@@ -0,0 +1,225 @@
+        CartHandlers.attachContext
+      ),
+    },
+  ],
+  [
+    CreateCartActions.attachRegion,
+    {
+      invoke: pipe(
+        getWorkflowInput(CartHandlers.attachRegion.aliases.Region),
+        CartHandlers.attachRegion
+      ),
+    },
+  ],
+  [
+    CreateCartActions.findOrCreateAddresses,
+    {
+      invoke: pipe(
+        {
+          invoke: [
+            getWorkflowInput(
+              CartHandlers.findOrCreateAddresses.aliases.Addresses
+            ).invoke,
+            {
+              from: CreateCartActions.attachRegion,
+              alias: CartHandlers.findOrCreateAddresses.aliases.Region,
+            },
+          ],
+        },
+        CartHandlers.findOrCreateAddresses
+      ),
+    },
+  ],
+  [
+    CreateCartActions.createCart,
+    {
+      invoke: pipe(
+        {
+          invoke: [
+            {
+              from: CreateCartActions.attachRegion,
+              alias: CartHandlers.createCart.aliases.Region,
+            },
+            {
+              from: CreateCartActions.attachContext,
+              alias: CartHandlers.createCart.aliases.Context,
+            },
+            {
+              from: CreateCartActions.findOrCreateCustomer,
+              alias: CartHandlers.createCart.aliases.Customer,
+            },
+            {
+              from: CreateCartActions.findOrCreateAddresses,
+              alias: CartHandlers.createCart.aliases.Addresses,
+            },
+          ],
+        },
+        CartHandlers.createCart
+      ),
+      compensate: pipe(
+        {
+          invoke: [
+            {
+              from: CreateCartActions.createCart,
+              alias: CartHandlers.removeCart.aliases.Cart,
+            },
+          ],
+        },
+        CartHandlers.removeCart
+      ),
+    },
+  ],
+  [
+    CreateCartActions.attachLineItems,
+    {
+      invoke: pipe(
+        {
+          invoke: [
+            getWorkflowInput(
+              CartHandlers.attachLineItemsToCart.aliases.LineItems
+            ).invoke,
+            {
+              from: CreateCartActions.createCart,
+              alias: CartHandlers.attachLineItemsToCart.aliases.Cart,
+            },
+          ],
+        },
+        CartHandlers.attachLineItemsToCart
+      ),
+    },
+  ],
+  [
+    CreateCartActions.retrieveCart,
+    {
+      invoke: pipe(
+        {
+          invoke: [
+            {
+              from: CreateCartActions.attachConfig,
+              alias: CartHandlers.attachConfig.aliases.Config,
+            },
+            {
+              from: CreateCartActions.createCart,
+              alias: CartHandlers.retrieveCart.aliases.Cart,
+            },
+          ],
+        },
+        CartHandlers.retrieveCart
+      ),
+    },
+  ],
+])
+
+WorkflowManager.register(Workflows.CreateCart, workflowSteps, handlers)
+
+type CreateCartWorkflowInput = Record<any, any>
+type CreateCartWorkflowOutput = Record<any, any>
```

Comment:
suggestion: same if not too complexe (in the sense of extracting all existing types)

Reply:
this is pretty hard, I'll skip this one for now.